DevEdu.API/DevEdu.Business/ValidationHelpers/TaskValidationHelper.cs

```diff
@@ -19,5 +24,21 @@ public void CheckTaskExistence(int taskId)
             if (task == default)
                 throw new EntityNotFoundException(string.Format(ServiceMessages.EntityNotFoundMessage, nameof(task), taskId));
         }
+
+        public void CheckTaskExistenceWithValidation(int taskId, int userId)
+        {
+            var groupsByTask = _taskRepository.GetGroupsByTaskId(taskId);
+            var groupsByUser = _userRepository.GetGroupsByUserId(userId);
+            List<GroupDto> grByT = new List<GroupDto>();
+            List<GroupDto> grByU = new List<GroupDto>();
+            foreach (var group in groupsByTask)
+                grByT.Add(group.Group);
+            foreach (var group in groupsByUser)
+                grByU.Add(group.Group);
+
+            var result = grByT.FirstOrDefault(gt => grByU.Any(gu => gu.Id == gt.Id));
+            if (result == default)
+                throw new AuthorizationException(string.Format(ServiceMessages.UserNotRelatedToTaskMessage, userId, taskId));
+        }
```

Comment:
1. you are not checking the task existence here, so you need to rename this method
2. GetGroupsByTaskId is a method that returns a list of GroupTaskDto (so this method MUST BE renamed), and you've got to call a method that returns a list of GroupDto by your taskId
3. GetGroupsByUserId that you have written returns the list of UserGroupDto, and you've got to call a method that returns a list of GroupDto by your userId
4. that means that UserGroupDto must be removed (and GetGroupsByUserId will have another implementation)
5. that means that you won't need those ugly foreach-loops
6. variables like grByT and grByU may be part of your PoC during the debug process, but can't be a part of commit to the main branch